main.rb

```diff
@@ -0,0 +1,122 @@
+module Enumerable
+  # my-each
+
+  def my_each
+    ind = 0
+
+    while ind != length
+      yield self[ind]
+      ind += 1
+    end
+
+    self
+  end
+
+  # my_each_with_index
+
+  def my_each_with_index
+    index = 0
+    my_each do |el|
+      yield(el, index)
+      index += 1
+    end
+
+    self
+  end
+
+  # my_select
+
+  def my_select
+    filtered_arr = []
+
+    while length != 0
+      yield(first) && filtered_arr.push(first)
+      shift
+    end
+
+    filtered_arr
+  end
+
+  # my_all
+
+  def my_all?
+    output = true
+
+    while length != 0
+      if !yield(first)
+        output = false
+        break
+      else
+        shift
+      end
+    end
+    output
+  end
```

Comment:
- [ ] should have working `my_all?` *identical and returns the same thing as ruby's [`all?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-all-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_all.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

- It shouldn't modify the original array.
- Your method doesn't consider the possibility when no block or argument is given. It should return true if the enumerable contains truthy values and false if it contains at least one falsy value.
- Your method doesn't consider the possibility when a class (String, Numeric, etc) is given as an argument `(some_enumerable.my_all?(String)).`
- Your method doesn't consider the possibility when a Regex is given as an argument `(some_enumerable.my_all?(/w/)).`
- Your method doesn't consider the possibility when a number is given as an argument `(some_enumerable.my_all?(5)).`